File: stack.py
class Node():
    def __init__(self, v, n):
        self.value = v
        self.next = n

class Stack():
    def __init__(self):
        self.head = None
        self.size = 0

    def push(self, v):
        self.head = Node(v, self.head)
        self.size += 1

    def pop(self):
        if self.head is None:
            return None
        tmp = self.head
        self.head = self.head.next
        self.size -= 1
        return tmp.value

    def top(self):
        if self.head is None:
            return None
        return self.head.value

    def empty(self):
        return self.head is None

    def size(self):
        return self.size 

    def p(self):
        n = self.head
        while n:
            print(n)
            n = n.next

def myStack(s): 
    stack = Stack()
    for i, c in enumerate(s):
        pos = i + 1
        if c == ')':
            if stack.empty() or stack.top()[0] != '(':
                return pos
            stack.pop()
        elif c == '}':
            if stack.empty() or stack.top()[0] != '{':
                return pos
            stack.pop()
        elif c == ']':
            if stack.empty() or stack.top()[0] != '[':
                return pos
            stack.pop()
        elif c == '(' or c == '{' or c == '[':
            pair = (c, pos)
            stack.push(pair)

    if stack.empty():
        return 'Success'
    else:
        return stack.top()[1]

# inp = input()
inp = 'foo(bar[i);'
print(myStack(inp))

File: stack.js
function node(value, next) {
    this.value = value;
    this.next = next;
}

function push(stack, value) {
    return new node(value, stack);
}

function pop(stack) {
    if (stack === null || stack === undefined) {
        return undefined;
    }

    return [stack.value, stack.next];
}

function parenthesis(str) {
    var stack = [];

    for (var i = 0; i < str.length; i++) {
        switch(str[i]) {
            case '(':
                // push(stack, str[i]);
                stack.push(str[i]);
                break;
            case '{':
                stack.push(str[i]);
                break;
            case '[':
                stack.push(str[i]);
                break;
            case ')':
                // var result = pop(stack);
                // value = result[0];
                // stack = result[1];
                if (stack.length === 0) {
                    return str.length;
                } else if (stack.pop() !== '(') {
                    return str.length;
                }
                break;
            case '}':
                if (stack.length === 0) {
                    return str.length;
                } else if (stack.pop() !== '{') {
                    return str.length;
                }
                break;
            case ']':
                if (stack.length === 0) {
                    return str.length;
                } else if (stack.pop() !== '[') {
                    return str.length;
                }
                break;
        }        
    }

    if (stack.length === 0) {
        return 'Success';
    } else {
        return str.length;
    }
}


var input = '{{[()]]';
//console.log(parenthesis(input));




function myStack(str) {
    var stack = [];

    for (var i = 0; i < str.length; i++) {
        if (stack[stack.length-1] === '(' && str[i] === ')') {
            stack.pop();
        } else if (stack[stack.length-1] === '{' && str[i] === '}') {
            stack.pop();
        } else if (stack[stack.length-1] === '[' && str[i] === ']') {
            stack.pop();
        } else {
            stack.push(str[i]);
        }  
    }

    if (stack.length === 0) {
        return 'Success';
    } else {
        console.log(stack);
        return str.length;
    }

}

console.log(myStack('({[]})'));